Added summary section

sygrinberg opened the pull request.

sygrinberg committed:
Added summary section

Elioravr commented on scripts/test.js:
```diff
@@ -12,20 +12,21 @@ const argv = process.argv.slice(2);
 // Watch unless on CI or in coverage mode
 if (!process.env.CI && argv.indexOf('--coverage') < 0) {
-  argv.push('--watch');
+    argv.push('--watch');
```
What exactly did happen here (in all the file)? 😅

sygrinberg replied:
It was an accident  :sweat_smile:
I fixed it.

Elioravr commented on src/components/SummarySection.js:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  render() {
+    console.log(this.props)
```
`console.log`

Elioravr commented on src/components/SummarySection.js:
```diff
@@ -0,0 +1,24 @@
+    return (
+      <div className="summary-container">
+        <label>Total:</label>
+        <span>{this.props.stepValues.summary.result.value}</span>
```
```
const {stepValues: summary} = this.props
const {result: value} = summary
```

Elioravr commented on src/store/reducers/selectors.js:
```diff
@@ -46,3 +46,33 @@ export const getCurrentStepTitle = store => {
   return title
 }
+
+// export const getWelcomeValue = store => {
```
Remove comments

Elioravr commented on src/pricing/calculations/roof.js:
```diff
@@ -0,0 +1,26 @@
+  }
+}
+
+export const getRoofParams = function(stepValues) {
```
What do you think about this syntax (please make sure you're using `es6` syntax instead of `es5` syntax, in the function declaration for example)?
```javascript


export const getRoofParams = ({which, slope, dimensions}) => {
  return  {
    tiles: tiles[which.result.value],
    slope: slope[slope.result.value],
    overHang: overHang[stepValues['over-hang'].result.value],
    dimensions: parseDimensions(dimensions.result)
  }
}
```

Elioravr commented on src/pricing/calculations/roof.js:
```diff
@@ -0,0 +1,26 @@
+import { slope } from '../content/roof/slope'
+import { overHang } from '../content/roof/overHang'
+
+function parseDimensions(dimensions) {
```
Don't you think this would be better (and notice the comment):
```javascript
const parseDimensions = ({floors, homeSquareFoot, garageSquareFoot}) => {
  return {
    floors: parseInt(floors.result.value, 10),
    homeSquareFoot: parseInt(homeSquareFoot.result.value, 10),
    garageSquareFoot: parseInt(garageSquareFoot.result.value || '1', 10)
    // Shouldn't it's default value be 0 instead of 1? We're adding it to the final
    // square foot, and not multiple it.
    // Maybe:
    //
    // garageSquareFoot: parseInt(garageSquareFoot.result.value || '0', 10)
    //
  }
}
```

Elioravr commented on src/pricing/content/roof/overHang.js:
```diff
@@ -0,0 +1,6 @@
+export const overHang = {
```
I very liked the concept, but maybe we can put this in 1 file for each category, instead of a full directory for each category? I don't see it grows so much, and i think that would be better.

Elioravr commented on src/steps/roofNextSteps.js:
```diff
@@ -40,5 +41,13 @@ export default [
     questions: roofContent['dimensions'].questions
+  },
+  {
+    sectionName: 'summary',
```
It's important to keep the titles clear, i'm sure you just didn't notice.
Just know that the main title and the sub title are taking place in some places in the app (the steps for example.
```javascript

  {
    sectionName: 'summary',
    sectionComponent: SummarySection,
    labels: {
      mainTitle: 'Price!',
      subTitle: 'This is the price it is going to cost for you'
    }
  }
```

Elioravr commented on src/store/reducers/stepReducer.js:
```diff
@@ -1,5 +1,9 @@
 import {steps} from 'steps'
+import {mainCalculator} from '../../pricing/calculations/main'
+
+console.log('steps')
```
`console.log`

Elioravr commented on src/store/reducers/stepReducer.js:
```diff
@@ -20,12 +26,19 @@ export default (state = initialState, action) => {
         }
       }
 
+      let summaryValues
```
I think we need to separate this to a `reducerHelper` file and put this inside a function there.

sygrinberg replied:
OK.
Can you please provide a brief example it should be done?

Elioravr commented on src/store/reducers/stepReducer.js:
```diff
@@ -20,12 +26,19 @@ export default (state = initialState, action) => {
       }
 
+      let summaryValues
+      if (steps.length > 2 && currentStepIndex === steps.length - 1) {
```
```javascript
// On the top of the file, after the imports:
const NUMBER_OF_BASE_STEPS = 2
...
// and then the condition should be
const isLastStep = steps.length > NUMBER_OF_BASE_STEPS && currentStepIndex === steps.length - 1
if (isLastStep) {
```

Elioravr reviewed (commented):
Coooollllll!! Looks great! I've just added some comments.. 👍

sygrinberg commented on scripts/test.js:
```diff
@@ -12,20 +12,21 @@ const argv = process.argv.slice(2);
 // Watch unless on CI or in coverage mode
 if (!process.env.CI && argv.indexOf('--coverage') < 0) {
-  argv.push('--watch');
+    argv.push('--watch');
```
It was an accident  :sweat_smile:

sygrinberg committed:
Fixed summary section rejects.

Elioravr commented on src/pricing/calculations/roof.js:
```diff
@@ -1,26 +1,31 @@
-import { overHang } from '../content/roof/overHang'
+import { tiles } from '../content/roof/categories'
+import { slope } from '../content/roof/categories'
+import { overHang } from '../content/roof/categories'
```
You can just do:
`import {tiles, slope, overHang} from '../content/roof/categories'`

Elioravr reviewed (commented):
Just fix the last thing and I'll approve this change

sygrinberg committed:
Added Exterior paint.

Elioravr commented on src/components/SummarySection.js:
```diff
@@ -9,9 +9,9 @@ export class SummarySection extends Component {
-    const { result: result } = summary
+    const { stepValues } = this.props
+    const { summary } = stepValues
+    const { result } = summary
```
Until now we write `const {stepValues} = this.props` (without spaces after and before the `{`, I think it's important that we'll write the same everywhere.

Elioravr commented on src/pricing/calculations/extPaint.js:
```diff
@@ -0,0 +1,35 @@
+import { floorsDiscount } from '../content/extPaint/categories'
+import { overHangRatio } from '../content/extPaint/categories'
+import { overHang } from '../content/extPaint/categories'
+import { windowPrice } from '../content/extPaint/categories'
```
Maybe you haven't seen my last comment (on the `calculations/roof.js` also), you can write:
`import {floorsDiscount, overHangRatio, overHang, windowPrice} from '../content/extPaint/categories'`

Elioravr commented on src/pricing/calculations/extPaint.js:
```diff
@@ -0,0 +1,35 @@
+  const surfaceWithGarage = discountedSurfaceSize + dimensions.garageSquareFoot
+  const overHangedPrice = params.overHang !== 'noOverHang' ? surfaceWithGarage * overHangRatio : surfaceWithGarage
+  const windowPriced = (params.windows * windowPrice) + overHangedPrice
+  return {value: windowPriced}
```
Maybe we should call it `{price: windowPriced}` so it would be clearer that we're talking about a price section, and not another section of questions. what do you think?

sygrinberg replied:
I think we should have both since value is a convention that we use, but adding price adds that its not another regular section.

Elioravr replied:
Do whatever you think.. I agree..

Elioravr commented on src/pricing/calculations/extPaint.js:
```diff
@@ -0,0 +1,35 @@
+  }
+}
+
+export const getParams = function({overHangValue, dimensions, windows}) {
```
`export const getParams = ({overHangValue, dimensions, windows}) => {`

Elioravr commented on src/pricing/calculations/extPaint.js:
```diff
@@ -0,0 +1,35 @@
+  }
+}
+
+export const calculate = function(stepValues) {
```
`export const calculate = (stepValues) => {`

Elioravr commented on src/pricing/calculations/extPaint.js:
```diff
@@ -0,0 +1,35 @@
+import { overHang } from '../content/extPaint/categories'
+import { windowPrice } from '../content/extPaint/categories'
+
+const parseDimensions = ({floors, homeSquareFoot, garageSquareFoot}) => {
```
Some function, of course, are going to repeat in other categories (such as `parseDimensions`), so maybe we should put it in a `calculationTools.js` file or something

Elioravr commented on src/pricing/calculations/roof.js:
```diff
@@ -10,7 +10,7 @@ const parseDimensions = ({floors, homeSquareFoot, garageSquareFoot}) => {
 }
 
-export const getRoofParams = function({whichValue, slopeValue, overHangValue, dimensions}) {
+export const getParams = function({whichValue, slopeValue, overHangValue, dimensions}) {
```
`export const getParams = ({whichValue, slopeValue, overHangValue, dimensions}) => {`

Elioravr commented on src/pricing/calculations/roof.js:
```diff
@@ -19,8 +19,8 @@ export const getRoofParams = function({whichValue, slopeValue, overHangValue, di
 
-export const calculateRoof = function(stepValues) {
-  const params = getRoofParams({
+export const calculate = function(stepValues) {
```
`export const calculate = (stepValues) => {`

Elioravr commented on src/steps/extPaintNextSteps.js:
```diff
@@ -0,0 +1,62 @@
+    sectionComponent: ButtonSection,
+    labels: {
+      mainTitle: "Which?",
+      subTitle: "Which kind of roof would you like to renovate in your house?"
```
It's not the roof they want to renovate, right? maybe the first step should be the paint choosing step, and this step should have the `hasBorderEffect`

sygrinberg replied:
You absolutely right!

Elioravr commented on src/steps/extPaintNextSteps.js:
```diff
@@ -0,0 +1,62 @@
+    sectionComponent: QuestionSection,
+    labels: {
+      mainTitle: "Windows",
+      subTitle: "How many windows would you like to paint?"
```
`"How many windows do you have in you're house?"`

Elioravr commented on src/steps/index.js:
```diff
@@ -26,5 +27,6 @@ export const steps = [
 export const nextSteps = {
-  roof: roofNextSteps
+  roof: roofNextSteps,
+  'exterior-paint': extPaintNextSteps
```
we have this strange thing that we do: we use strings as keys in objects just because we want to write in `dash-case`, but we can write in `camelCase` and it would be lot easier everywhere in the application.

sygrinberg replied:
yes, I agree with that.
I will apply it on the 'over-hang' thing too

Elioravr replied:
Great!

Elioravr commented on src/steps/productContents/exteriorPaint.js:
```diff
@@ -0,0 +1,130 @@
+    ]
+  },
+  'dimensions': {
+    questions: [
```
All the steps and the questions that is repeating here (from the roof), we can put in a separate file and it would be easier to menage them.

sygrinberg replied:
you right. 
but lets discuss about it with Shay, and see what's common and then do it.

Elioravr commented on src/steps/stepsContent.js:
```diff
@@ -18,7 +18,7 @@ export const stepsContent = {
         { value: "roof", text: 'Roof' },
         { value: "windows", text: 'Windows' },
-        { value: "exteriorPaint", text: 'Exterior Paint' },
+        { value: "exterior-paint", text: 'Exterior Paint' },
```
If we change it from `'exterior-paint'` to `exteriorPaint` than we need to change it here also.

Elioravr commented on src/pricing/calculations/main.js:
```diff
@@ -1,10 +1,13 @@
-import { calculateRoof } from './roof'
+import { calculate as calculateRoof } from './roof'
+import { calculate as calculateExtPaint } from './extPaint'
```
COOLLLL!!!!

Elioravr reviewed (changes requested):
I finished. Take a look, they're all semantic changes. Just try to stick with the `es6` syntax and try not to repeat code. You've done great job!

sygrinberg committed:
Fixed comments.
Changed the steps and nextSteps file structure.
Added currency formatter to the SummarySection.

sygrinberg committed:
Changed windows content.
Placed dimensions step content to be common.
